### ISSUE TITLE
[FEATURE REQUEST] salt dig module: dig.x

### ISSUE THREAD

**Issue by matthewsaum** [saltstack/salt#62275]
**Is your feature request related to a problem? Please describe.**
I love the use of salt modules, both in my jinja templates and for regular system administration, but I find the ability to do a reverse-lookup of an IP somewhat lacking. 

**Describe the solution you'd like**
The salt module `dig` has a handful of useful functions already, the most-used being `dig.A`, so I suggest trying to get `dig.x` to also work.

**Describe alternatives you've considered**
I've been running `cmd.shell host 1.2.3.4` and other such workarounds, but a module would be great.

**Additional context**

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by matthewsaum** [comment]
I added a block in a local module in my environment. Just felt it ought to be pushed upward for proper integration.
```
def X(host, nameserver=None):
    '''
    Return the PTR record for ``host``.

    Always returns a list.

    CLI Example:

    .. code-block:: bash

        salt ns1 dig.X 1.2.3.4
    '''
    dig = ['dig', '+short', '-x', six.text_type(host) ]

    if nameserver is not None:
        dig.append('@{0}'.format(nameserver))

    cmd = __salt__['cmd.run_all'](dig, python_shell=False)

    if cmd['retcode'] != 0:
        log.warning(
            'dig returned exit code \'{0}\'. Returning empty list as '
            'fallback.'.format(
                cmd['retcode']
            )
        )
        return []

    return [i for i in cmd['stdout'].split('\n')]
```

**Comment by nicholasmhughes** [comment]
@matthewsaum , that's great! Would you be willing to put that functionality into a PR? It would only need a couple of basic unit tests See https://github.com/saltstack/salt/pull/61992 for ideas on how to make that happen and feel free to ping me with any additional questions.

**Comment by matthewsaum** [comment]
@nicholasmhughes  Happy to try to figure it out. I'll see if I can get it working and get back to you before end of day Friday.

**Comment by matthewsaum** [comment]
I realize now having read up on your links that I have to be a part of a working group to do so. This... makes sense. Given our teams extensive use of salt, I'm completely on board with this.

**Comment by matthewsaum** [comment]
Please to report I've got documentation and a local dev environment running. I'll poke it and get a proper test suite run and have a merge request incoming.

**Comment by matthewsaum** [comment]
So I learned a bit more about things while doing that first attempted Pull Request. New PR should be popping up shortly.

**Comment by nicholasmhughes** [comment]
Closed by #62314
